CRM/Case/BAO/CaseType.php

```diff
@@ -190,7 +190,20 @@ public static function convertDefinitionToXML($name, $definition) {
       $xmlFile .= "</CaseRoles>\n";
     }
 
+    if (array_key_exists('restrictActivityAsgmt', $definition)) {
+      $xmlFile .= "<RestrictActivityAsgmt>" . $definition['restrictActivityAsgmt'] . "</RestrictActivityAsgmt>\n";
+    }
+
+    if (!empty($definition['activityAsgmtGrps'])) {
+      $xmlFile .= "<ActivityAsgmtGrps>\n";
+      foreach ($definition['activityAsgmtGrps'] as $value) {
+        $xmlFile .= "<Group>$value</Group>\n";
+      }
+      $xmlFile .= "</ActivityAsgmtGrps>\n";
+    }
```

Comment:
You could save some space in the xml file and prevent errors by skipping this section if not needed. E.g. have the conditional on 197 read

    if (!empty($definition['activityAsgmtGrps']) && $definition['restrictActivityAsgmt'] == 2) {